src/bindings/c/include/openvino/c/ov_shape.h

```diff
@@ -22,15 +22,24 @@ typedef struct {
 } ov_shape_t;
 
 /**
- * @brief Init a shape object, allocate space for its dimensions.
+ * @brief Initialize a fully shape object, allocate space for its dimensions and set its content.
  * @ingroup shape
  * @param rank The rank value for this object, it should be more than 0(>0)
+ * @param dims The dimensions data for this shape object, it's size should be equal to rank.
  * @param ov_status_e a status code.
  */
-OPENVINO_C_API(ov_status_e) ov_shape_init(ov_shape_t* shape, int64_t rank);
+OPENVINO_C_API(ov_status_e) ov_shape_init(ov_shape_t* shape, int64_t rank, int64_t* dims);
 
 /**
- * @brief Free a shape object's internal memory
+ * @brief Initialize a shape object, only allocate space for its dimensions.
+ * @ingroup shape
+ * @param rank The rank value for this object, it should be more than 0(>0).
+ * @param ov_status_e a status code.
+ */
+OPENVINO_C_API(ov_status_e) ov_shape_init_dimension(ov_shape_t* shape, int64_t rank);
```

Comment:
Maybe we can have one only one method ov_shape_create() and put a `NULL` in case if user don't want to fill dimensions:
`ov_shape_create(shape, 4, NULL);

Reply:
ok

Reply:
done